inference-engine/src/gna_plugin/gna_graph_tools.hpp

```diff
@@ -436,7 +436,45 @@ inline void CNNNetSwapLayers(InferenceEngine::CNNLayerPtr lhs,
     lhs->outData.front()->setDims(rhs->outData.front()->getDims());
 }
 
+/**
+* @brief changes the Tensor Desctiption if data by created a new one with correct description and replacing original one
+*/
+inline DataPtr CNNDataChangeTensorDescription(DataPtr data, TensorDesc td) {
+    auto new_dataPtr = std::make_shared<Data>(data->getName() + "_reshaped", td);
+    getInputTo(new_dataPtr) = getInputTo(data);
+    auto creatorLayer = getCreatorLayer(data).lock();
+    getCreatorLayer(new_dataPtr) = creatorLayer;
+
+    auto idx = -1;
+    for (int i=0; i < creatorLayer->outData.size(); i++) {
+        if (areEqualDatas(data, creatorLayer->outData[i])) idx = i;
```

Comment:
Should there be "break" when corresponding outData is found?

Reply:
I think CNNLayerFindInsDataIdxes does exactly what you trying to do here.

Reply:
No, CNNLayerFindInsDataIdxes can be used to find indexes of input data and not output data